docs/howto/migrate/migration-guide.md

```diff
@@ -0,0 +1,591 @@
+| zio.Console.Service.live | zio.Console.ConsoleLive |
+| zio.System.Service.live  | zio.System.SystemLive   |
+| zio.Random.Service.live  | zio.Random.RandomLive   |
+
+1. All ZIO services moved to the `zio` package:
+
+    | ZIO 1.x               | ZIO 2.x                       |
+    |-----------------------|-------------------------------|
+    | zio.blocking.Blocking | [Removed](#blocking-service)  |
+    | zio.clock.Clock       | zio.Clock                     |
+    | zio.console.Console   | zio.Console                   |
+    | zio.random.Random     | zio.Random                    |
+    | zio.system.System     | zio.System                    |
+
+    And their live implementations renamed and moved to a new path:
+
+    | ZIO 1.x                  | ZIO 2.x                 |
+    |--------------------------|-------------------------|
+    | zio.Clock.Service.live   | zio.Clock.ClockLive     |
+    | zio.Console.Service.live | zio.Console.ConsoleLive |
+    | zio.System.Service.live  | zio.System.SystemLive   |
+    | zio.Random.Service.live  | zio.Random.RandomLive   |
+
+
+2. In ZIO 2.0 all type aliases like `type Logging = Has[Logging.Service]` removed. So we should explicitly use `Has` wrappers when we want to specify dependencies on ZIO services.
+
+So instead of writing `ZLayer[Console with Clock, Nothing, ConsoleLogger]`, we should write `ZLayer[Has[Console] with Has[Clock], Nothing, Has[ConsoleLogger]]`.
+
+### Blocking Service
+
+Since there is rarely a need to use a separate blocking thread pool, ZIO 2.0 created _one global blocking pool_ removed the Blocking service from `ZEnv` and the built-in services.
+
+All blocking operations were moved to the `ZIO` data type:
+
+| ZIO 1.x                 | ZIO 2.x |
+|-------------------------|---------|
+| zio.blocking.Blocking.* | ZIO.*   |
+
+We can also provide a user-defined blocking executor in ZIO 2.x with the `Runtime#withBlockingExecutor` operator that constructs a new `Runtime` with the specified blocking executor.
+
+### Console Service
+
+Method names in the _Console_ service were renamed to the more readable names:
+
+| ZIO 1.x     | ZIO 2.x        |
+|-------------|----------------|
+| putStr      | print          |
+| putStrErr   | printError     |
+| putStrLn    | printLine      |
+| putStrLnErr | printLineError |
+| getStrLn    | readLine       |
+
+## Improvements
+
+### Composable Zips
+
+In ZIO 2.x, when we are zipping together different effects:
+- `Tuple`s are not nested.
+- `Unit`s do not contribute to the output.
+
+Assume we have these effects
+
+```scala mdoc:silent:nest
+val x1: Task[Int]     = Task.succeed(???)
+val x2: Task[Unit]    = Task.succeed(???)
+val x3: Task[String]  = Task.succeed(???)
+val x4: Task[Boolean] = Task.succeed(???)
+```
+
+In ZIO 1.x, the output of zipping together these effects are nested:
+
+```scala
+val zipped = x1 <*> x2 <*> x3 <*> x4
+// zipped: ZIO[Any, Throwable, (((Int, Unit), String), Boolean)] = zio.ZIO$FlatMap@3ed3c202
+```
```

Comment:
Task.succeed actually returns a `UIO`. Perhaps we can do:

```scala
val x1: UIO[Int]     = ZIO.succeed(???)
val x2: UIO[Unit]    = ZIO.succeed(???)
val x3: UIO[String]  = ZIO.succeed(???)
val x4: UIO[Boolean] = ZIO.succeed(???)
```

In ZIO 1.x, the output of zipping together these effects are nested:

```scala
val zipped = x1 <*> x2 <*> x3 <*> x4
// zipped: ZIO[Any, Throwable, (((Int, Unit), String), Boolean)] = zio.ZIO$FlatMap@3ed3c202
```
```